// kysely-migration-cli/dist/cli.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.main = void 0;
const _1 = require(".");
const kysely_1 = require("kysely");
const path_1 = __importDefault(require("path"));
const fs_1 = require("fs");
const pg_1 = require("pg");
/**
 * Load Dotenv if the module exists.
 */
try {
    require('dotenv').config();
}
catch (_a) { }
const { DATABASE_URL } = process.env;
if (!DATABASE_URL) {
    const hint = `Error: DATABASE_URL not set.

Please specify DATABASE_URL to run this CLi. Try the following:
  - Run \`DATABASE_URL=postgres://user:password@host:port/database && npm run kysely-migration-cli\`
  - Place .env file containing \`DATABASE_URL=postgres://user:password@host:port/database\`
`;
    console.log(hint);
    process.exit(1);
}
const db = new kysely_1.Kysely({
    dialect: new kysely_1.PostgresDialect({
        pool: new pg_1.Pool({
            connectionString: DATABASE_URL,
        }),
    }),
});
const migrator = new kysely_1.Migrator({
    db,
    provider: new kysely_1.FileMigrationProvider({
        path: path_1.default,
        fs: fs_1.promises,
        migrationFolder: path_1.default.resolve(process.cwd(), './migrations'),
    }),
});
function main() {
    (0, _1.run)(db, migrator);
}
exports.main = main;
